app/dashboards/statistic_dashboard.rb

```diff
@@ -0,0 +1,145 @@
+require "administrate/custom_dashboard"
+
+class StatisticDashboard < Administrate::CustomDashboard
+  resource "Statistics"
+
+  class Filter
+    TIME_FORMATS = {
+      day: "%d %B %Y",
+      month: "%B %Y",
+      year: "%Y"
+    }.freeze
+
+    attr_reader :name, :reference_time, :time_period, :column
+
+    def initialize(name:, reference_time:, time_period:, column: :created_at)
+      @name = name
+      @reference_time = reference_time
+      @time_period = time_period
+      @column = column
+    end
+
+    def apply(scope)
+      scope.where(column => start_time..end_time)
+    end
+
+    def to_s
+      reference_time.strftime(TIME_FORMATS.fetch(time_period))
+    end
+
+    private
+
+    def start_time
+      reference_time.public_send("beginning_of_#{time_period}")
+    end
+
+    def end_time
+      reference_time.public_send("end_of_#{time_period}")
+    end
+  end
+
+  class AllFilter < Filter
+    def initialize(name:)
+      super(name: name, reference_time: nil, time_period: nil)
+    end
+
+    def apply(scope)
+      scope
+    end
+
+    def to_s
+      name.to_s.titleize
+    end
+  end
+
+  DEFAULT_FILTER = Filter.new(name: :today, reference_time: Time.current, time_period: :day)
+
+  FILTERS = [
+    AllFilter.new(name: :all_time),
+    DEFAULT_FILTER,
+    Filter.new(name: :yesterday, reference_time: 1.day.ago, time_period: :day),
```

Comment:
`reference_time` needs to be evaluated in runtimes. It shouldn't be in boot time.